OloEngine/src/OloEngine/Renderer/Commands/CommandBucket.cpp

```
@@ -278,29 +362,94 @@ namespace OloEngine
+            // Check against max instances limit
+            if (totalTransforms > m_Config.MaxMeshInstances)
             {
-                instancedCmd->transforms[instancedCmd->instanceCount++] = sourceCmd->transform;
-                return true;
+                OLO_CORE_WARN("CommandBucket::TryMergeCommands: Max instances ({}) reached", m_Config.MaxMeshInstances);
+                return false;
             }
 
-            return false;
+            u32 newOffset = frameBuffer.AllocateTransforms(totalTransforms);
+            if (newOffset == UINT32_MAX)
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)